tests/test_lightningd.py

```diff
@@ -195,21 +196,38 @@ def fund_channel(self, l1, l2, amount):
         l1.daemon.wait_for_log('-> CHANNELD_NORMAL')
         l2.daemon.wait_for_log('-> CHANNELD_NORMAL')
 
-    def pay(self, lsrc, ldst, amt, label=None):
+    def pay(self, lsrc, ldst, amt, label=None, async=False):
         if not label:
             label = ''.join(random.choice(string.ascii_letters + string.digits) for _ in range(20))
 
         rhash = ldst.rpc.invoice(amt, label)['rhash']
         assert ldst.rpc.listinvoice(label)[0]['complete'] == False
 
-        routestep = {
-            'msatoshi' : amt,
-            'id' : ldst.info['id'],
-            'delay' : 5,
-            'channel': '1:1:1'
-        }
-        lsrc.rpc.sendpay(to_json([routestep]), rhash)
-        assert ldst.rpc.listinvoice(label)[0]['complete'] == True
+        def call_pay():
+            routestep = {
+                'msatoshi' : amt,
+                'id' : ldst.info['id'],
+                'delay' : 5,
+                'channel': '1:1:1'
+            }
+            lsrc.rpc.sendpay(to_json([routestep]), rhash, async=False)
+
+        t = threading.Thread(target=call_pay)
```

Comment:
I generally prefer using futures, since they're shorter and can easily be joined later. They'll also capture any eventual exception so we can avoid the backtrace :wink: